build.gradle

```diff
@@ -26,8 +26,27 @@ repositories {
     }
 }
 
+sourceSets {
+    create("compatibilityApi")
+    create("develocityCompatibility")
+    create("enterpriseCompatibility")
+}
+
 dependencies {
-    compileOnly('com.gradle:develocity-gradle-plugin:3.17-rc-1')
+    compatibilityApiCompileOnly(gradleApi())
+
+    develocityCompatibilityCompileOnly(gradleApi())
+    develocityCompatibilityCompileOnly('com.gradle:develocity-gradle-plugin:3.17-rc-1')
+    develocityCompatibilityImplementation(sourceSets.compatibilityApi.output)
+
+    enterpriseCompatibilityCompileOnly(gradleApi())
+    enterpriseCompatibilityCompileOnly('com.gradle:gradle-enterprise-gradle-plugin:3.16.2')
+    enterpriseCompatibilityImplementation(sourceSets.compatibilityApi.output)
+
+    api(sourceSets.compatibilityApi.output)
```

Comment:
I'm curious about why we are exposing the `enterpriseCompatibility` classes and `develocityCompatibility` classes as `api`. I understand why the `compatibilityApi` is marked as `api` but why does the consumer also need to know about the implementation details here?

Reply:
Fair, changed it in https://github.com/gradle/common-custom-user-data-gradle-plugin/pull/280/commits/1e1951475bfa44958ea75c337d6d078876d6568c